compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -345,8 +345,24 @@ class Definitions {
     newSymbol(ScalaPackageClass, tpnme.IMPLICITkw, EmptyFlags, TypeBounds.empty).entered
   def ImplicitScrutineeTypeRef: TypeRef = ImplicitScrutineeTypeSym.typeRef
 
-
-  @tu lazy val ScalaPredefModule: Symbol = ctx.requiredModule("scala.Predef")
+  @tu lazy val ScalaPredefModule: Symbol = {
+    val mod = ctx.requiredModule("scala.Predef")
+    mod.infoOrCompleter match {
```

Comment:
Maybe I should not do this if it is under Scala2 mode
```scala
if (!ctx.scala2Mode) {
  mod.infoOrCompleter match {
  ...
```

Reply:
This will not work as accessing the language module requires loading the symbol of Predef.

Reply:
An alternative would be to only enable this change if the `-migration` setting is not set. In that case the conversion would still be triggered and would emit the deprecation waring.